### DISCUSSION TITLE
Possibility to have a client sided NT SetSynchronizedProperties

### DISCUSSION THREAD

**Discussion by hesham-akmal** [FirstGearGames/FishNet#303] [Approved Changes or Features]
For a long time I used the NT.SetSynchronizedProperties at runtime on clients to prevent NT pos syncing of a non owned player, when this player starts a cinematic animation. The new "canSend" value in "public void SetSynchronizedProperties(SynchronizedProperty value)" prevents client from setting this value (I noticed it after updating to 3.0.0). So I had to make SetSynchronizedPropertiesInternal public.

**Comment by FirstGearGames** [comment]
I checked and it is public. This is of course an old discussion; I assume you're all set?

**Reply by hesham-akmal** [reply]
The method public is "SetSynchronizedProperties(SynchronizedProperty value)" which does not allow the method call from a spectator (the canSend bool will be false since IsOwner is false), now I know this was intentional. It's just that in my case, in my game there is many cinematics where I implemented client-sided movement freeze, so I want all players to quickly locally freeze a non owned player's movement. So my workaround is setting "SetSynchronizedPropertiesInternal" method to public. What do you think

**Comment by FirstGearGames** [comment]
Sorry for the delay on this.
Changing the synchronized properties without the proper permissions can cause errors on the client changing them.
If anything it might be best to add a 'freeze interpolation' API or something that lets you pause movements until you resume.

This is kind of what you're doing already but your approach is much riskier.

**Reply by hesham-akmal** [reply]
oh no worries.
Got it! Thanks a lot for the advice, I will definitely change it to freezing interpolation then :)

**Comment by FirstGearGames** [comment]
Topic mismatch for conclusion.
Add feature to temporarily pause/turn off interpolation; this will not synchronize.

When disabled interpolations will be dropped except for latest. When unpaused the object will snap to the latest received data.

**Comment by FirstGearGames** [comment]
This appears to have been added quite awhile ago!
